### ISSUE TITLE
Mint-Y in Flatpak runtime does not update

### ISSUE THREAD

**Issue by ccprog** [linuxmint/mint-themes#416]
No idea if this is the right place, but my impression is that this is not an issue with mint-update or flatpak itself. Nonetheless, possibly related: https://github.com/linuxmint/mintupdate/issues/779

In Update Manager (mintUpdate 5.9.6), I get presented the following:

![flatpak](https://user-images.githubusercontent.com/5325416/208921422-9a5988f3-52ce-42cc-8b1f-f72a4ea026b2.png)

Confirming the update, I do not get asked about root rights, and after a short query period, the same entry is presented again, without any update happening.

Trying to update from the command line, I did this:

```
claus@copernikus:~$ flatpak info --system org.gtk.Gtk3theme.Mint-Y 
Mint-Y Gtk Theme - Mint-Y theme

     Kennung: org.gtk.Gtk3theme.Mint-Y
         Ref: runtime/org.gtk.Gtk3theme.Mint-Y/x86_64/3.22
 Architektur: x86_64
       Zweig: 3.22
    Ursprung: flathub
    Sammlung: org.flathub.Stable
Installation: system
 Installiert: 318,0 kB

      Commit: 8530793683478255edff3d0b88c80b196be10dc3edb3a414aa6…
      Parent: a2a4a6456cc6f99d678f1943a4678cbb1457ea5dd71d64155da…
     Subject: Fix theme dependency (55c071b2)
        Date: 2019-09-21 09:40:09 +0000
claus@copernikus:~$ sudo flatpak update --system org.gtk.Gtk3theme.Mint-Y
[sudo] Passwort für claus:           
Looking for updates…


        KENNUNG                  Zweig Op Remote  Download
 1. [✗] org.gtk.Gtk3theme.Mint-Y 3.22  u  flathub 1,0 kB / 96,5 kB

Error: Während des Holens von runtime/org.gtk.Gtk3theme.Mint-Y/x86_64/3.22 von der entfernten Quelle flathub: opcode set-read-source: Opening content object 013c8321e9949f63909b99d22484fa39a9405ee705649b7b23556595ccbfb42e: fgetxattr(user.ostreemeta): Keine Daten verfügbar
Updates complete.
Fehler: There were one or more errors
```

**Comment by mtwebster** [comment]
Try running `flatpak repair` and try again. This makes it sound as though there's something corrupt in your system flatpak installation.

**Comment by ccprog** [comment]
The error message remains the same.

Since the report of flatpak repair is quite endless, I hoe these few last lines give an impression.

```
Deleting ref flathub:runtime/org.gtk.Gtk3theme.Mint-Y/x86_64/3.22 due to invalid objects

Checking remotes...
Pruning objects
Erasing .removed
Reinstalling removed refs
Installing runtime/org.gtk.Gtk3theme.Mint-Y-Aqua/x86_64/3.22
Installing runtime/org.gtk.Gtk3theme.Mint-Y/x86_64/3.22
Error: Failed to installieren org.gtk.Gtk3theme.Mint-Y: Während des Holens von runtime/org.gtk.Gtk3theme.Mint-Y/x86_64/3.22 von der entfernten Quelle flathub: opcode set-read-source: Opening content object 013c8321e9949f63909b99d22484fa39a9405ee705649b7b23556595ccbfb42e: Couldn't find file object '013c8321e9949f63909b99d22484fa39a9405ee705649b7b23556595ccbfb42e'
claus@copernikus:~$ sudo flatpak update --system org.gtk.Gtk3theme.Mint-Y
Looking for updates…


        KENNUNG                  Zweig Op Remote  Download
 1. [✗] org.gtk.Gtk3theme.Mint-Y 3.22  u  flathub 1,0 kB / 96,5 kB

Error: Während des Holens von runtime/org.gtk.Gtk3theme.Mint-Y/x86_64/3.22 von der entfernten Quelle flathub: opcode set-read-source: Opening content object 013c8321e9949f63909b99d22484fa39a9405ee705649b7b23556595ccbfb42e: Couldn't find file object '013c8321e9949f63909b99d22484fa39a9405ee705649b7b23556595ccbfb42e'
Updates complete.
Fehler: There were one or more errors

**Comment by mtwebster** [comment]
Ah sorry, just found this: https://github.com/flatpak/flatpak/issues/4116

Try `sudo flatpak repair`

**Comment by ccprog** [comment]
I **did** run it with root privileges.

**Comment by ccprog** [comment]
You were right that somehow my installation was corrupt. But `sudo flatpak repair` was not enough to resolve the problem. I ended up completely uninstalling `flatpak`, removing `/var/lib/flatpak` and starting from scratch again.

_For anyone else trying to do this: this also removes the flathub reference and GPG key signature, and it does not get restored when re-installing `flatpak`. I had to copy it from a timeshift backup._
